packages/flutter/lib/src/material/button_style.dart

```diff
@@ -435,3 +435,33 @@ class _LerpShapes implements MaterialStateProperty<OutlinedBorder?> {
     return ShapeBorder.lerp(resolvedA, resolvedB, t) as OutlinedBorder?;
   }
 }
+
+/// How the icon in any `button.icon` should be placed, where `button` could be:
+///
+/// * [TextButton]
+/// * [ElevatedButton]
+/// * [OutlinedButton]
+///
+/// In buttons that have an icon, we generally tend to use a `Row` widget
+/// to be able to put multiple widgets in a horizontal way, aka `Row`.
+///
+/// Hover over enum values to be able to see more deailed description
+/// on what do they do.
+enum ButtonIconAlignment {
+  /// Places the icon to the `start` of the button which means the
+  /// first child in the `Row`.
+  ///
+  /// The `Row` will be like this if you choose to use `start` which is
```

Comment:
avoid "you" in the API docs if possible.

Reply:
thanks, I have removed the whole block because as requested it has an implementation that could change and be used by other types of buttons